src/marvin/utilities/openai.py

```diff
@@ -41,19 +41,24 @@ def get_openai_client(
         )
 
         if not api_key:
-            raise ValueError(
-                inspect.cleandoc(
-                    """
-                    OpenAI API key not found! Marvin will not work properly without it.
-                    
-                    You can either:
-                        1. Set the `MARVIN_OPENAI_API_KEY` or `OPENAI_API_KEY` environment variables
-                        2. Set `marvin.settings.openai.api_key` in your code (not recommended for production)
+            if (
+                api_url := marvin.settings.openai.base_url
+            ) is None or api_url.startswith("https://api.openai.com"):
+                raise ValueError(
+                    inspect.cleandoc(
+                        """
+                        OpenAI API key not found! Marvin will not work properly without it.
                         
-                    If you do not have an OpenAI API key, you can create one at https://platform.openai.com/api-keys.
-                    """
+                        You can either:
+                            1. Set the `MARVIN_OPENAI_API_KEY` or `OPENAI_API_KEY` environment variables
+                            2. Set `marvin.settings.openai.api_key` in your code (not recommended for production)
+                            
+                        If you do not have an OpenAI API key, you can create one at https://platform.openai.com/api-keys.
+                        """
```

Comment:
# Error Handling and Messaging
The error message is very clear, guiding the user on how to resolve the issue. However, it assumes the absence of an API key as the root problem, which might not always be the case (e.g., network issues, and misconfigurations).